src/openzaak/notificaties/mixins.py

```diff
@@ -0,0 +1,46 @@
+import logging
+from io import StringIO
+
+from .models import FailedNotification
+
+
+class FailedNotificationFilter(logging.Filter):
+    def filter(self, record):
+        if record.levelname == "WARNING":
+            return True
+        return False
+
+
+class FailedNotificationMixin:
+    def notify(self, status_code, data, instance=None):
+        stream = StringIO()
+        console = logging.StreamHandler(stream=stream)
+        logger = logging.getLogger("vng_api_common.notifications.viewsets")
+        logger.addHandler(console)
+
+        logger.addFilter(FailedNotificationFilter())
```

Comment:
This doesn't seem like the right approach to me.

Why not create a log handler that puts stuff in a model? And register it via settings?

This whole mixin concept seems inappropriate.

Let's discuss with @sergei-maertens

Reply:
Indeed, I had something like https://pypi.org/project/django-db-logger/ in mind instead